client/cli/src/lib.rs

```diff
@@ -942,8 +953,14 @@ pub fn fill_import_params<C, G, E>(
 
 	config.wasm_method = cli.wasm_method.into();
 
+	let execution = if is_dev {
+		cli.execution_strategies.execution.or(Some(ExecutionStrategy::Native))
+	} else {
+		cli.execution_strategies.execution
+	};
+
 	let exec = &cli.execution_strategies;
-	let exec_all_or = |strat: ExecutionStrategy| exec.execution.unwrap_or(strat).into();
+	let exec_all_or = |strat: ExecutionStrategy| execution.unwrap_or(strat).into();
```

Comment:
This will default to `execution` even if `strat` is ovewritten in CLI, I thought you wanted to use `strat` and only default to `execution`, no?

Reply:
Seems like we need to test whether exec was explicitly set or if it's a default as part of `is_dev` being true.